src/components/_classes/component/Component.js

```diff
@@ -2723,6 +2723,7 @@ export default class Component extends Element {
     // Handle all cases when calculated values should not fire.
     if (
       (this.options.readOnly && !this.options.pdf && !this.component.calculateValue) ||
+      (this.options.readOnly && this.options.pdf) ||
```

Comment:
There are cases where customers are generating PDF's using the POST apis directly to the PDF server. So in this case, this would break those customers use case.